### PR TITLE
feat(embedly): Remove basic auth and query string data for embedly requests

### LINKED ISSUE #196: Remove basic auth and query string for embedly requests
I'm a little bit concerned about us leaking really sensitive stuff in urls the embedly requests, so I'm wondering if we should see how good the results are if we remove basic auth/query strings (just for embedly). We can still show the full URL locally, just not send it out over the wire.

It could be that this is *too* restrictive, but let's play with this/try something out.

### PR DESCRIPTION
**Pull request by pdehaan**
r? @k88hudson 

Fixes #196

### CODE REVIEW

**Comment by pdehaan** [comment]
Here's a screenshot of my local data. The full, unedited URL is still displayed in the activity stream, but you can see the embedly proxy is only sent the pre-approved list of query params.

new_tab

File: content-src/lib/utils.js [hunk 1]
```diff
@@ -1,16 +1,53 @@
+var urlParse = require("url-parse");
```
Comment (k88hudson) [inline content-src/lib/utils.js]: `const` please! we should add an eslint rule for that

File: content-src/lib/utils.js [hunk 2]
```diff
@@ -1,16 +1,53 @@
...
     }
     return url.replace(/^((https?:)?\/\/)?(www\.)?/i, "").toLowerCase();
+  },
+
+  sanitizeUrl(site) {
+    if (!site) {
+      return "";
+    }
+
+    const ALLOWED_QUERY_PARAMS = new Set(["id", "p", "q", "query", "s", "search", "sitesearch"]);
```
Comment (k88hudson) [inline content-src/lib/utils.js]: should we store these at the top of the file perhaps?

**Comment by k88hudson** [comment]
Looks good to me! Just a couple of nits

File: content-src/lib/utils.js (hunk 1 above)
Comment (pdehaan) [inline content-src/lib/utils.js]: I'll add the ESLint rule in https://github.com/mozilla/activity-streams/pull/218#discussion_r54442320.

**Pull request merged**